api/tests/functional-tests/backend/core/test_llm_clients.py

```diff
@@ -122,19 +126,19 @@
     ]
 }```"""
 
-HALLUCINATION_AGREEMENT_VALID_VERDICTS = """```json
+HALLUCINATION_VALID_VERDICTS = """```json
 {
     "verdicts": [
         {
-            "verdict": "yes"
+            "verdict": "no"
         },
 
         {
-            "verdict": "no",
-            "reason": "The text and context mention disagree on when Abraham Lincoln was born."
+            "verdict": "yes",
+            "reason": "The text and context disagree on when Abraham Lincoln was born."
         },
         {
-            "verdict": "no",
+            "verdict": "yes",
```

Comment:
Why did these tests change?

Reply:
I changed these from agreement verdicts to contradiction verdicts, so all the tests needed to swap 'yes' to 'no and vice versa. 

I made this change because DeepEval used an instruction asking for agreement verdicts, but their instruction was very confusing. They tell the LLM to respond 'yes' if the prediction text agrees with the contexts, but 'no' if only if the prediction text directly contradicts the contexts. I don't know if agreement is really the exact opposite of contradiction. I felt like this left the case ambiguous when the text is unrelated to the contexts. 

The hallucination metric is the proportion of contradictions, so I felt like it would be better to just ask the LLM for contradiction verdicts. That's why I flipped from agreement verdicts to contradiction verdicts. 

Unfortunately, this is an artefact of DeepEval having low quality instructions. I would look at RAGAS here if I could, but RAGAS doesn't have a hallucination metric.